layers/dns.go

```diff
@@ -464,16 +465,32 @@ func (rr *DNSResourceRecord) String() string {
 	return "..."
 }
 
+func decodeCharacterStrings(data []byte) ([][]byte, error) {
+	var strings [][]byte
+	end := len(data)
+	for index, index2 := 0, 0; index != end; index = index2 {
+		index2 = index + 1 + int(data[index])
+		if index2 > end {
+			return nil, errors.New("Insufficient data for a <character-string>")
+		}
+		strings = append(strings, data[index+1:index2])
```

Comment:
Are TXT records guaranteed to be at least one char?  If so, maybe check for that here?